src/sensor/ping360.cpp

```diff
@@ -93,8 +93,8 @@ void Ping360::handleMessage(const ping_message& msg)
         _transmit_duration = deviceData.transmit_duration();
         _sample_period = deviceData.sample_period();
         _transmit_frequency = deviceData.transmit_frequency();
-        deviceData.number_of_samples();
 
+        _data.resize(deviceData.number_of_samples());
```

Comment:
Why do we need to dynamically resize the buffer? It appears to be an unnecessary operation that will hurt performance. I think we should allocate the maximum expected size one time in the constructor.

Reply:
We prealocate 2k memory, resize will only change the `size()` return of the class.

Reply:
[QByteArray::resize()](https://doc.qt.io/qt-5/qbytearray.html#resize) looks like it might do a reallocation every time. @tcanabrava do you know if this is an expensive operation?

Reply:
I made a mistake `_data` here is QVector, not QByteArray: https://doc.qt.io/qt-5/qvector.html#resize